tests/plenary/path_spec.lua

```diff
@@ -389,54 +415,105 @@ describe("Path", function()
+    end)
+
+    it("should throw on invalid args", function()
+      local before = env.new_path { touch = true }
+      assert.errors(function()
+        before:rename { new_name = "" }
+      end)
+      assert.errors(function()
+        before:rename {}
+      end)
+      assert.errors(function()
+        before:rename()
+      end)
+      assert.is.True(before:exists())
+    end)
+
+    it("should throw if old name doesn't exist", function()
+      local before, after = env.new_path { touch = false }, env.new_path { touch = false }
+      assert.errors(function()
+        before:rename { new_name = after }
+      end)
     end)
 
     it("can move to parent dir", function()
-      local p = Path:new "some_random_filename.lua"
-      assert(pcall(p.touch, p))
-      assert(p:exists())
-
-      assert(pcall(p.rename, p, { new_name = "../some_random_filename.lua" }))
-      assert.are.same(vim.loop.fs_realpath(Path:new("../some_random_filename.lua"):absolute()), p:absolute())
-
-      p:rm()
+      local before, after = env.new_path { filename = "random_file" }, env.new_path { filename = "../random_file" }
+      assert.is.False(before:exists())
+      assert.is.False(after:exists())
+      before:touch()
+      assert.is.True(before:exists())
+      before:rename { new_name = after }
+      assert.is.False(before:exists())
+      assert.is.True(after:exists())
+    end)
+
+    it("should throw on rename to existing filename", function()
+      local before, after = env.new_path { touch = true }, env.new_path { touch = true }
+      assert.errors(function()
+        before:rename { new_name = after }
+      end)
+    end)
+
+    it("shouldn't throw on rename to same filename", function()
+      local before = env.new_path { touch = true }
+      assert.does.Not.error(function()
+        before:rename { new_name = before }
+      end)
+    end)
+
+    -- BUG: introduced in f71ee45 (#90)
+    it("should handle . or .. or ./ or ../ prefix", function()
+      for _, pre in ipairs { ".", "..", "./", "../" } do
+        local before, after =
+          env.new_path { filename = "random_file" }, env.new_path { filename = pre .. "random_file2" }
+        assert.is.False(before:exists())
+        assert.is.False(after:exists())
+        before:touch()
+        assert.is.True(before:exists())
+        before:rename { new_name = after }
+        assert.is.False(before:exists())
+        assert.is.True(after:exists())
+      end
     end)
 
-    it("cannot rename to an existing filename", function()
-      local p1 = Path:new "a_random_filename.lua"
-      local p2 = Path:new "not_a_random_filename.lua"
-      assert(pcall(p1.touch, p1))
-      assert(pcall(p2.touch, p2))
-      assert(p1:exists())
-      assert(p2:exists())
-
-      assert(not pcall(p1.rename, p1, { new_name = "not_a_random_filename.lua" }))
-      assert.are.same(p1.filename, "a_random_filename.lua")
-
-      p1:rm()
-      p2:rm()
+    -- BUG: introduced in f71ee45 (#90)
```

Comment:
Is this a workaround, bugfix or what do you mean herre on linking to #90?

Reply:
Sorry I can add details to this if you want me to? Would you like me to remove the `BUG:` tag too? Or I can just remove it completely?

I put this in just for clarity and to describe the test (i.e. the reason for the test) because it tests for something specific that was introduced in a single commit/pr. If the bug didn't exist, this test probably wouldn't either.

Reply:
The usual approach is to 1. create an issue, if needed and 2.  link to the issue. That way potential contributors can see the issue, discuss it (if necessary) and just fix it.
This also prevents the introduction of expalnaation of unnecessary complex context into the code etc.